recipes/parallel-preprocessor/meta.yaml

```
@@ -0,0 +1,80 @@
+# manually update or generated from template
+package:
+    name: parallel-preprocessor
+    version: '0.3'
+source:
+    # calc check sum by command sha256sum
+    sha256: 0f472735203b16e092686003e50e25d2e6dbdeef43da8f14177d6bcb4db86634
+    url: https://github.com/ukaea/parallel-preprocessor/releases/download/v0.3/parallel-preprocessor.zip
+
+build:
+    # number should be generated, or manually update, YYMM pattern here, must be 0 for the first PR
+    number: '0'
+    string: oct_2020
+    skip: true  # [py < 38 or osx]
+
+# libGL needs Core Dependency Tree (CDT) packages
+# https://conda-forge.org/docs/maintainer/knowledge_base.html#special-dependencies
+requirements:
+    run_constrained:
+        - __osx >={{ MACOSX_DEPLOYMENT_TARGET|default("10.9") }}  # [osx]
+    build:
+        - python
+        - cmake
+        - git
+        - ninja
+        - make  # [unix]
+        - {{ compiler("cxx") }}
+        - {{ cdt('mesa-libgl-devel') }}      # [linux]
+        - {{ cdt('mesa-dri-drivers') }}      # [linux]
+        - {{ cdt('mesa-libegl-devel') }}     # [linux]
+        - {{ cdt('libselinux') }}            # [linux]
+        - {{ cdt('libxdamage') }}            # [linux]
+        - {{ cdt('libxfixes') }}             # [linux]
+        - {{ cdt('libxxf86vm') }}            # [linux]
+        - {{ cdt('libxcb') }}                # [linux]
+        - {{ cdt('libxext') }}               # [linux]
+        - {{ cdt('xorg-x11-server-xvfb') }}  # [linux]
+        - {{ cdt('libxau') }}                # [linux]
+        - {{ cdt('libxi-devel') }}           # [linux]
+        - occt >=7.4
+        - tbb
+        - freetype
+        - tbb-devel
+        - pybind11
```

Most of these don't belong in `build`. Also see https://conda-forge.org/docs/maintainer/knowledge_base.html#libgl
```suggestion
        - cmake
        - git
        - make  # [unix]
        - {{ compiler("cxx") }}
        - {{ cdt('mesa-libgl-devel') }}  # [linux]
        - {{ cdt('mesa-dri-drivers') }}  # [linux]
        - {{ cdt('libselinux') }}  # [linux]
        - {{ cdt('libxdamage') }}  # [linux]
        - {{ cdt('libxxf86vm') }}  # [linux]
        - {{ cdt('libxext') }}     # [linux]
```

---

thanks I have no in-depth understanding on the build env,  it is just pure luck and days of googling to make it working.